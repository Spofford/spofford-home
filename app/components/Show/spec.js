import React from "react"
import expect from "expect"
import { shallow } from "enzyme"

import { Show } from "./"

describe("<Show />", () => {
  it("should render", () => {
    const renderedComponent = shallow(
      <Show />
    )
    expect(renderedComponent.is("div")).toEqual(true)
  })
})
