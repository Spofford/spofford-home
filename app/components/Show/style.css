.show .fullHeader .container {
  background-color: #585858;
  box-shadow:none;
}

.show .hero {
  background-color: #585858;
  height:calc(100% - 65px);
  margin-top:65px;
}

.show .hero h3 {
  color: white;
}

.show > .fullHeader h1, .show > .fullHeader > .container .fa-bars, .show > .fullHeader > .container .menu-label, .show > .fullHeader .search-container.hidden .fa-search, .show .cancelTrigger, .show .cancelTrigger, .show .search-container .link-container span, .show .search-container .link-container span {
   color:white;
 }
