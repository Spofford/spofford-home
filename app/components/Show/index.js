import React from "react"
const ReactMarkdown = require('react-markdown')
import * as contentful from 'contentful'
import style from "./style.css"
import { connect } from "react-redux"
import cssModules from "react-css-modules"
import { default as Header } from "../Header"
import { Link } from 'react-router-dom'
import FontAwesome from "react-fontawesome";

export class Show extends React.Component {
  state = {
    model: {}
  }

  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })

  componentDidMount() {
    /* window.scrollTo(0, 0); */
    this.fetchModel().then(this.setModel);
  }

  fetchModel = () => this.client.getEntry('6mwsq2p1ewaw0sOay2uCEw')

  setModel = response => {
    this.setState({
      model: response.fields
    })
  }

  render() {
    return (
      <div className="show">
        <Header headerStart={true} />
        <div className="hero">
          <div className="copy-container">
            <h2>{this.state.model.head}</h2>
            <h3>Bringing together circular design, regionality, and human-centeredness to imagine New England furniture's future.</h3>
          </div>
        </div>
        <div className="body-container">

          <div className="copy-container">
            <h2>{this.state.model.subhead2}</h2>
            <ReactMarkdown source={this.state.model.bodyText2} />
            <Link to='/signup'>start your submission<FontAwesome name='chevron-right' /></Link>
            <hr />
          </div>

          <div className="copy-container">
            <h2>{this.state.model.subhead3}</h2>
            <ReactMarkdown source={this.state.model.bodyText3} />
            <Link to='/signup'>start your submission<FontAwesome name='chevron-right' /></Link>
          </div>

          <div className="copy-container">
            <h2>{this.state.model.subhead4}</h2>
            <ReactMarkdown source={this.state.model.bodyText4} />
            <Link to='/signup'>start your submission<FontAwesome name='chevron-right' /></Link>
          </div>

          <div className="copy-container">
            <h2>{this.state.model.subhead4}</h2>
            <ReactMarkdown source={this.state.model.bodyText4} />
            <Link to='/signup'>start your submission<FontAwesome name='chevron-right' /></Link>
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user
})

export default connect(mapStateToProps)(cssModules(Show, style))
