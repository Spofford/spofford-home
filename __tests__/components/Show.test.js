import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Show from '../../app/components/Show';

describe('<Show />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Show />);
      // const component = wrapper.dive();

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('calls setModel on mount', async () => {
      const wrapper = shallow(<Show />);
      const spy = jest.spyOn(wrapper.instance(), "setModel");
      wrapper.instance().forceUpdate();
      wrapper.instance().componentDidMount();

      const data = await wrapper.instance().fetchModel();
      expect(spy).toHaveBeenCalled();
    })
  });
});
