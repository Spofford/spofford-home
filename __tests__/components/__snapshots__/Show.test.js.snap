// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> render() renders the component 1`] = `
<div
  className="show"
>
  <Connect(Header)
    headerStart={true}
  />
  <div
    className="hero"
  >
    <div
      className="copy-container"
    >
      <h2 />
      <h3>
        Bringing together circular design, regionality, and human-centeredness to imagine New England furniture's future.
      </h3>
    </div>
  </div>
  <div
    className="body-container"
  >
    <div
      className="copy-container"
    >
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        replace={false}
        to="/signup"
      >
        start your submission
        <FontAwesome
          name="chevron-right"
        />
      </Link>
      <hr />
    </div>
    <div
      className="copy-container"
    >
      <h2 />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        replace={false}
        to="/signup"
      >
        start your submission
        <FontAwesome
          name="chevron-right"
        />
      </Link>
      <hr />
    </div>
    <div
      className="copy-container"
    >
      <h2 />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        replace={false}
        to="/signup"
      >
        start your submission
        <FontAwesome
          name="chevron-right"
        />
      </Link>
    </div>
    <div
      className="copy-container"
    >
      <h2 />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        replace={false}
        to="/signup"
      >
        start your submission
        <FontAwesome
          name="chevron-right"
        />
      </Link>
    </div>
  </div>
</div>
`;
